src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -611,28 +629,30 @@ class Cluster[F[_]](
     logThread(
       withMetric(
         {
-          dao.nodeConfig.whitelisting
-            .get(hp.host)
-            .map(PeerClientMetadata(hp.host, hp.port, _))
-            .map {
-              client =>
-                apiClient.sign
-                  .getRegistrationRequest()
-                  .run(client)
-                  .flatMap { registrationRequest =>
-                    for {
-                      _ <- pendingRegistration(hp.host, registrationRequest)
-                      prr <- pendingRegistrationRequest
-                      response <- apiClient.sign.register(prr).run(client)
-                    } yield response
-                  }
-                  .handleErrorWith { err =>
-                    logger.error(s"registration request failed: $err") >>
-                      dao.metrics.incrementMetricAsync[F]("peerGetRegistrationRequestFailed") >>
-                      err.raiseError[F, Unit]
-                  }
-            }
-            .getOrElse(logger.error(s"unknown peer, not whitelisted"))
+          createAndSetNewOwnToken >>
+            dao.nodeConfig.whitelisting
+              .get(hp.host)
+              .map(PeerClientMetadata(hp.host, hp.port, _))
+              .map {
+                client =>
+                  apiClient.sign
+                    .getRegistrationRequest()
+                    .run(client)
+                    .flatMap { registrationRequest =>
+                      for {
+                        _ <- logger.warn(s"I GOT registrationRequest: $registrationRequest")
+                        _ <- pendingRegistration(hp.host, registrationRequest)
+                        prr <- pendingRegistrationRequest
+                        response <- apiClient.sign.register(prr).run(client)
+                      } yield response
+                    }
+                    .handleErrorWith { err =>
+                      logger.error(s"registration request failed: $err") >>
```

Comment:
Change it to `logger.error("Registration request failed")(err)`

Reply:
Sure I can refactor that.